src/StreamrClient.js

```diff
@@ -245,7 +259,7 @@ export default class StreamrClient extends EventEmitter {
         }
     }
 
-    subscribe(optionsOrStreamId, callback, legacyOptions) {
+    async subscribe(optionsOrStreamId, callback, legacyOptions) {
```

Comment:
This completely breaks existing clients! :( Ideally `subscribe` would keep returning the `Subscription` instead of a `Promise` (which `async` will do). It should be possible if we don't needlessly call `getSessionToken()` here and pass it to the `Subscription` (see my comments in `Subscription`)